### DISCUSSION TITLE
[Bloblang] New Method `must_array`

### DISCUSSION THREAD

**Discussion by gramian** [redpanda-data/connect#2082] [Ideas]
Dear All,

I would like to propose a new **method** for bloblang. This method provisionally called `must_array` behaves as follows:

* if the underlying type is an array, `must_array` returns its input (identity)
* if the underlying type is not an array, `must_array` wraps its input in an array

Examples:
```
root = this.something.must_array()

# in: { "something": "Hi Benthos" }
# out: ["Hi Benthos"]

#in: { "something": [1,2,3,4] }
#out: [1,2,3,4]
```

Currently, one can emulate this behavior in bloblang:
```
root = if this.type() == "array" { this } else { [this] }
```

It could be discussed, how this function would behave for `null`: for example an empty array could be returned(?).

Why is this useful?

I am working with `xml` messages, which when parsed into `json` yield objects or strings if only one instance of a tag appears, and an array if multiple instances are found in one node. To process such results further with the same machinery these parsed tags need thave to be arrays, so `map_each`, `fold`, `filter` can be applied equally.

A real-life example for such an `xml` format is [`mods`](https://www.loc.gov/standards/mods/).

The emulated function can be made a `map` to be reused, but if I understand correctly, for example, it would need to be imported or redefined in every mapping of every branch in a workflow.

Thank you for considering my proposal!

PS: This came up in the "questions" Discord channel.

**Comment by mihaitodor** [comment]
Hey @gramian, I guess you could also use `this.with("something").values().flatten()` as a workaround.

**Reply by gramian** [reply]
This is also a good tip. Thank you.
